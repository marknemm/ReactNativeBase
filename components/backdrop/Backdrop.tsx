import Animated, { FadeIn, FadeOut } from 'react-native-reanimated';
import type { BackdropProps } from './Backdrop.interfaces';
import { useStyles } from './Backdrop.styles';

/**
 * Component for a modal backdrop.
 *
 * @param props The component {@link BackdropProps}.
 * @returns The {@link Backdrop} component.
 */
const Backdrop: React.FC<BackdropProps> = (props) => {
  const { isVisible = true, onPress } = props;
  const styles = useStyles(props);

  return isVisible && (
    <Animated.View
      accessibilityRole={onPress ? 'button' : undefined}
      entering={FadeIn}
      exiting={FadeOut}
      onTouchEnd={onPress}
      style={styles.backdrop}
      testID="rnb-backdrop"
    />
  );
};

export default Backdrop;
export type * from './Backdrop.interfaces';
