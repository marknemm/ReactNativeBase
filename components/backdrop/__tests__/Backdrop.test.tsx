import Backdrop from '@components/backdrop/Backdrop';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen } from '@testing-library/react-native';

describe('<Backdrop />', () => {
  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <Backdrop />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when not visible', () => {
      render(
        <Backdrop isVisible={false} />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
