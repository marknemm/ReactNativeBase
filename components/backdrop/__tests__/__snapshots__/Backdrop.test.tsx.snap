// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> snapshots renders correctly 1`] = `
<View
  collapsable={false}
  entering={[Function]}
  exiting={[Function]}
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
  style={
    {
      "backgroundColor": "#393e4233",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  testID="rnb-backdrop"
/>
`;

exports[`<Backdrop /> snapshots renders correctly when not visible 1`] = `null`;
